dark/graphics.py

```diff
@@ -703,3 +721,41 @@ def scatterAlign(seq1, seq2, window=7):
     plt.title('Dot plot using window size %i\n(allowing no mis-matches)' %
               window)
     plt.show()
+
+
+def plotAAProperties(sequence, propertyNames, showPlot=True):
+    """
+    Plot amino acid property values for a sequence.
+
+    @param sequence: An C{AARead} (or a subclass) instance.
+    @param propertyNames: An iterable of C{str} property names (each of which
+        must be a key of a key in the dark.aa.PROPERTY_DETAILS dict).
+    @param showPlot: If C{True}, display the plot. Passing C{False} is useful
+        in testing.
+    @raise ValueError: If an unknown property is given in C{propertyNames}.
+    @return: The return value from calling dark.aa.propertiesForSequence:
+        a C{dict} keyed by (lowercase) property name, with values that are
+        C{list}s of the corresponding property value according to sequence
+        position.
+    """
+    MISSING_AA_VALUE = -1.1
+    propertyValues = propertiesForSequence(sequence, propertyNames,
+                                           missingAAValue=MISSING_AA_VALUE)
+    if showPlot:
+        legend = []
+        x = np.arange(0, len(sequence))
+
+        for index, propertyName in enumerate(propertyValues):
+            color = TABLEAU20[index]
+            plt.scatter(x, propertyValues[propertyName], color=color)
+            legend.append(patches.Patch(color=color, label=propertyName))
+
+        plt.legend(handles=legend, loc=(0, 1.1))
+        plt.xlim(0, len(sequence))
+        plt.ylim(min(MISSING_AA_VALUE, -1.1), 1.1)
```

Comment:
I put that in there to be defensive. If someone one day changes the value of `MISSING_AA_VALUE` to something higher or lower, the code will continue to do the right thing. So the person making the change doesn't have to worry about reading the rest of the code.